### ISSUE TITLE
Add Blosc2 Compression to ImageData

### ISSUE THREAD

**Issue by EmilDohne** [EmilDohne/PhotoshopAPI#10]
Currently our program uses an excessive amount of memory as it stores all the image data in an uncompressed manner which leads to big files (a focus of what this API should be able to handle) taking up loads of memory.

An example file which unfortunately cannot be shared is ~6GB on disk but takes up >80GB of memory when decompressing it in release mode. The file contains 170 Image layers (and some more mask layers) and has a bit depth of 16 with a resolution of 9124x6082

The goal of this change would be to bring memory usage on par with, or slightly above what photoshop files store but without sacrificing on performance too much.

**Comment by EmilDohne** [comment]
From initial testing this seems very promising, the 82 gigabytes of in-memory data get compressed to about 6 gigabyte while also offering a ~2x speed increase due to less memory transfer

**Comment by EmilDohne** [comment]
While checking on the compression I also implemented AVX2 SIMD support for byteswapping. 

Additionally I found that even on my 16 core AMD 5950x I was only experiencing a ~3x speedup in wall time from the current parallel implementation of the ChannelImageData. This turned out to likely be due to the parallel loops accessing data at vastly different locations. The implementation has been improved a lot now, cutting out on loads of `memcpy()` which brought processing time down from about 2.5 mins on all cores to 45 secs on 4 cores while keeping memory usage around 10 gigabytes. Roughly 1/3 of what photoshop consumes when opening the file.

Unfortunately, what is holding back performance still is the fact that the zip compressed Image data can not trivially be parallelized. I have brought down the rest of the processing as much as possible and will continue to do so gradually
